asf_hp_installer_chatbot/pipeline/heat_pump_chatbot_post_embedding.py

```diff
@@ -0,0 +1,302 @@
+index_name = "chatbot-onboarding"
+model_name = "text-embedding-ada-002"
+output_file = "/outputs/embedding/vector_embeddings_nibe_f2040_231844-5.pkl"
+chatbot_prompt = """
+[Chatbot Name]: Heat Pump Companion
+[Objective]: To provide friendly, accurate, and helpful information on heat pump installation, maintenance, and troubleshooting to a professional heat pump installer.
+[Tone]: Conversational and positive, with a focus on being helpful and reassuring to the user.
+[Knowledge Base]: The NIBE 2040 installer manual.
+[RAG Model Instructions]:
+1. [Retrieve]: When a user query is received, first identify key terms related to heat pumps (e.g., installation, types, efficiency, troubleshooting) and use them to retrieve the most relevant documents from the knowledge base.
+2. [Generate]: Based on the information retrieved, generate a response that is tailored to the user's query. Ensure the response is easy to understand, avoids technical jargon unless necessary, and provides actionable advice or clear information.
+3. [Tone Adjustment]: Apply a conversational tone to the generated response, aiming to mimic a friendly expert providing advice. Use phrases that reassure the user, such as "Great question!", "Here's what you need to know,", or "I'm here to help with your heat pump questions."
+4. [Contextual Relevance]: Ensure the response is directly relevant to the user's query, providing specific information about heat pumps as requested. If the query is about installation, focus on providing guidance about selecting the right heat pump from your [Knowledge Base], installation process, and tips for ensuring efficient operation.
+[User Interaction Examples]:
+- [User Query]: "What's the best heat pump for a small home?"
+- [RAG Response]: "Great question! For a small home, you'll want a heat pump that's efficient and sized appropriately to save on energy costs while keeping your space comfortable. A ductless mini-split system is often a good choice. They're versatile and can be more energy-efficient for smaller spaces."
+- [User Query]: "How often do I need to service my heat pump?"
+- [RAG Response]: "Regular maintenance is key to keeping your heat pump running smoothly. It's recommended to have it serviced at least once a year by a professional. This helps ensure efficiency and prolongs the life of your system."
+"""
+
+
+# Read in CSV file with vector embeddings and metadata
+def get_vector_embeddings_df(
+    output_file: str = "/outputs/embedding/vector_embeddings_nibe_f2040_231844-5.pkl",
+) -> pd.DataFrame:
+    """
+    Loads and returns a DataFrame of vector embeddings from a pickle file.
+
+    Args:
+        output_file (str, optional): The path to the pickle file relative to the project directory.
+        Defaults to '/outputs/embedding/vector_embeddings_nibe_f2040_231844-5.pkl'.
+
+    Returns:
+        pd.DataFrame: A DataFrame containing vector embeddings.
+    """
+    return pd.read_pickle(f"{PROJECT_DIR}{output_file}")
```

Comment:
```suggestion
    return pd.read_pickle(os.path.join(PROJECT_DIR, output_file))
```
would be less error prone i think